### ISSUE TITLE
OuterTypeFilename reports false positive on JEP 512 compact source files

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#19909]
I have read check documentation: https://checkstyle.org/checks/misc/outertypefilename.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
vivek@Viveks-MacBook-Air issue-pkg % 
cat > Hello.java << 'EOF'
void main() {
    System.out.println("Hello!");
}

class Helper {}
EOF

vivek@Viveks-MacBook-Air issue-pkg % 
cat > config.xml << 'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="OuterTypeFilename"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air issue-pkg % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar /Users/vivek/checkstyle/target/checkstyle-13.5.0-SNAPSHOT-all.jar \
    -c config.xml Hello.java

Starting audit...
[ERROR] /private/tmp/issue-pkg/Hello.java:5:1: The name of the outer type and the file do not match. [OuterTypeFilename]
Audit done.
Checkstyle ends with 1 errors.
```

### Describe what you expect in detail.

This is a JEP 512 compact source file. In a compact source file, the real outer type is an implicit unnamed class that wraps all top-level methods, fields, and any classes declared at the file root, see [JEP 512 §"Top-level methods and fields"](https://openjdk.org/jeps/512). The class Helper {} declaration is therefore semantically nested inside the implicit class, not an outer type, so the "outer type name must match the filename" rule does not apply.

OuterTypeFilename currently treats any CLASS_DEF at the top of the file as an outer type, ignoring the presence of top-level methods/fields that turn the compilation unit into a compact source file. This produces a false positive in the example above.

The check should detect that the compilation unit is a compact source file (i.e. has top-level METHOD_DEF/VARIABLE_DEF as direct children of COMPILATION_UNIT) and skip enforcement in that case.
